src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs

```diff
@@ -0,0 +1,428 @@
+            }
+        }
+        #endregion
+
+        #region PythonCodeCompletionProviderCommon public methods implementations
+
+        /// <summary>
+        /// Generate completion data for the specified text, while import the given types into the
+        /// scope and discovering variable assignments.
+        /// </summary>
+        /// <param name="code">The code to parse</param>
+        /// <param name="expand">Determines if the entire namespace should be used</param>
+        /// <returns>Return a list of IExternalCodeCompletionData </returns>
+        public override IExternalCodeCompletionData[] GetCompletionData(string code, bool expand = false)
+        {
+            IEnumerable<PythonCodeCompletionDataCore> items = new List<PythonCodeCompletionDataCore>();
+
+            if (!PythonEngine.IsInitialized)
+            {
+                PythonEngine.Initialize();
+                PythonEngine.BeginAllowThreads();
+            }
+
+            try
+            {
+                using (Py.GIL())
+                {
+                    if (code.Contains("\"\"\""))
+                    {
+                        code = StripDocStrings(code);
+                    }
+
+                    UpdateImportedTypes(code);
+                    UpdateVariableTypes(code);  // Possibly use hindley-milner in the future
+
+                    // If expand param is true use the entire namespace from the line of code
+                    // Else just return the last name of the namespace
+                    string name = expand ? GetLastNameSpace(code) : GetLastName(code);
+                    if (!String.IsNullOrEmpty(name))
+                    {
+                        try
+                        {
+                            // Attempt to get type using naming
+                            Type type = expand ? TryGetTypeFromFullName(name) : TryGetType(name);
+
+                            // CLR type
+                            if (type != null)
+                            {
+                                items = EnumerateMembers(type, name).Select(x => new PythonCodeCompletionDataCore(x.Item1, x.Item2, x.Item3, x.Item4, this));
+                            }
+                            // Variable assignment types
+                            else if (VariableTypes.TryGetValue(name, out type))
+                            {
+                                items = EnumerateMembersForPyObj(PyObject.FromManagedObject(type), name).Select(x => new PythonCodeCompletionDataCore(x.Item1, x.Item2, x.Item3, x.Item4, this));
+                            }
+                            else
+                            {// Python objects from the Scope
+                                var mem = LookupMember(name);
+                                if (mem == null && expand)
+                                {
+                                    // There is an issue with calling Dir() on nested PyObjects (not all members are discovered)
```

Comment:
perhaps this is related to the preload=true for this scope?

Reply:
I will look into it

Reply:
I tried it with preload=true...and a bunch of other tricks...but nothing works.
Maybe it's a pythonnet bug...(I could not track it down though...)